bootstrap.rkt

```diff
@@ -0,0 +1,476 @@
+                                                           (fancy-cases cases
+                                                             (generate-quasiquote
+                                                               (let [failure-cc
+                                                                     (lambda ()
+                                                                       ,inner-case)]
+                                                                 ,inner-cases)
+                                                               stx))))])]
+                                               (>>= (fancy-cases cases
+                                                      (generate-quasiquote
+                                                        (failure-cc)
+                                                        stx))
+                                                 (lambda (outer-cases)
+                                                   (pure (generate-quasiquote
+                                                           (let [,stx-name ,stx-expr]
+                                                             (let [failure-cc
+                                                                   (lambda ()
+                                                                     (syntax-error (list-syntax
+                                                                                     ('"syntax-case: the input"
+                                                                                      ,stx-name
+                                                                                      '"does not match any of the following patterns"
+                                                                                      ',(map car cases))
+                                                                                     stx)
+                                                                                   stx))]
+                                                               ,outer-cases))
+                                                           stx)))))))))
+                                     (syntax-error (list-syntax
+                                                     ('"syntax-case:"
+                                                      keywords
+                                                      '"does not look like a list of keywords."
+                                                      '"did you forget a () between the input and the cases?")
+                                                     stx))))))])))]
+                   [fancy-quasiquote
+                    (lambda (stx)
+                      (generate-syntax-case fancy-quasiquote stx ()
+                        [(_ ,pat)
+                         (let [stx-name (generate-quasiquote
+                                          ',(replace-loc pat 'here)
+                                          'here)]
```

Comment:
That incantation allowed me to get error messages to point to the callers `fancy-quasiquote` rather than to the source code of `fancy-quasiquote`, but I don't fully understand the magic behind the incantation.